pkg/oneinch/limitorder/extension.go

```diff
@@ -0,0 +1,169 @@
+package limitorder
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	eMath "github.com/ethereum/go-ethereum/common/math"
+)
+
+const (
+	totalOffsetSlots     = 8
+	offsetSlotSizeInBits = 32
+	offsetLength         = totalOffsetSlots * offsetSlotSizeInBits / 8
+)
+
+// Extension represents the extension data of a 1inch order.
+// This is copied from
+// nolint: lll
+// https://github.com/1inch/limit-order-sdk/blob/999852bc3eb92fb75332b7e3e0300e74a51943c1/src/limit-order/extension.ts#L6
+type Extension struct {
+	MakerAssetSuffix []byte
+	TakerAssetSuffix []byte
+	MakingAmountData []byte
+	TakingAmountData []byte
+	Predicate        []byte
+	MakerPermit      []byte
+	PreInteraction   []byte
+	PostInteraction  []byte
+	CustomData       []byte
+}
+
+func (e Extension) HasMakerPermit() bool {
+	return len(e.MakerPermit) > 0
+}
+
+func (e Extension) IsEmpty() bool {
+	return len(e.getConcatenatedInteractions()) == 0
+}
+
+func (e Extension) Encode() string {
+	interactionsConcatenated := e.getConcatenatedInteractions()
+	if len(interactionsConcatenated) == 0 {
+		return hexutil.Encode(interactionsConcatenated)
+	}
+
+	offsetsBytes := e.getOffsets()
+
+	b := new(bytes.Buffer)
+
+	b.Write(eMath.PaddedBigBytes(offsetsBytes, offsetLength))
+	b.Write(interactionsConcatenated)
+	b.Write(e.CustomData)
+
+	return hexutil.Encode(b.Bytes())
+}
+
+func (e Extension) interactionsArray() [totalOffsetSlots][]byte {
+	return [totalOffsetSlots][]byte{
+		e.MakerAssetSuffix,
+		e.TakerAssetSuffix,
+		e.MakingAmountData,
+		e.TakingAmountData,
+		e.Predicate,
+		e.MakerPermit,
+		e.PreInteraction,
+		e.PostInteraction,
+	}
+}
+
+func (e Extension) getConcatenatedInteractions() []byte {
+	builder := new(bytes.Buffer)
+	for _, interaction := range e.interactionsArray() {
+		builder.Write(interaction)
+	}
+	return builder.Bytes()
+}
+
+func (e Extension) getOffsets() *big.Int {
```

Comment:
Offsets should have 32 bytes as the code on blockchain said. `*big.Int` make no sense here.

```suggestion
func (e Extension) getOffsets() common.Hash {
```